src/converter/one/convert/JfrConverter.java

```diff
@@ -21,31 +21,39 @@
 public abstract class JfrConverter extends Classifier {
     protected final JfrReader jfr;
     protected final Arguments args;
+    private final IEventAggregator eventAggregator;
     protected Dictionary<String> methodNames;
 
     public JfrConverter(JfrReader jfr, Arguments args) {
+        boolean leakDetection = args.nativemem && args.leak;
+
         this.jfr = jfr;
         this.args = args;
+
+        IEventAggregator agg = new EventAggregator(args.threads, args.total, args.lock ? 1e9 / jfr.ticksPerSec : 1.0);
+        this.eventAggregator = leakDetection ? new MallocLeakAggregator(agg) : agg;
     }
 
     public void convert() throws IOException {
         jfr.stopAtNewChunk = true;
         while (jfr.hasMoreChunks()) {
             // Reset method dictionary, since new chunk may have different IDs
             methodNames = new Dictionary<>();
-            convertChunk();
+            convertChunk(collectEvents());
+            eventAggregator.resetChunk();
         }
-    }
 
-    protected abstract void convertChunk() throws IOException;
+        eventAggregator.finish();
+        convertChunk(eventAggregator);
+        eventAggregator.resetChunk();
```

Comment:
From the design perspective, it looks strange that `convertChunk/resetChunk` are called again after `finish`.
I understand they are needed only for `MallocLeakAggregator` to process the final filtered list, but I think this should be somehow hidden in `finish`.

Reply:
Moving them to their own method, to make the intent clearer.